### DISCUSSION TITLE
Difference between setting the home offsets in Configuration.h vs on the machine using the LCD in advanced settings?

### DISCUSSION THREAD

**Discussion by cristian-rivera** [Jyers/Marlin#967] [Q&A] [answered]
I was playing around with different configurations today and decided to build my own firmware with my machine specific settings baked in.

When loading up both the firmware from the official repo releases and mine, after choosing "Home All Axis's", the probe ends up in different locations!

With my home offsets defined in the code, the probe ended up further back and to the right on the board whereas with the same exact home offsets set on the LCD, the probe was closer to the front / left of the bed. Is this expected behavior?

* Front / left is the area on the machine above the SD card slot

These were my only changes compared to https://github.com/Jyers/Marlin/blob/Ender_3_V2_Extensible_UI/Configuration%20Files/E3V2%20Templates/BLTouch-3x3/Configuration.h

```
 // Printer's unique ID, used by some programs to differentiate between machines.
 // Choose your own or use a service like https://www.uuidgenerator.net/version4
@@ -800,7 +800,7 @@
  * Override with M92
  *                                      X, Y, Z, E0 [, E1[, E2...]]
  */
+#define DEFAULT_AXIS_STEPS_PER_UNIT   { 80, 80, 400, 139 }
-#define DEFAULT_AXIS_STEPS_PER_UNIT   { 80, 80, 400, 93 }

 /**
  * Default Max Feed Rate (mm/s)
@@ -1237,8 +1237,8 @@
 #define Y_BED_SIZE 230

 // Travel limits (mm) after homing, corresponding to endstop positions.
+#define X_MIN_POS -5
+#define Y_MIN_POS -10
-#define X_MIN_POS 0
-#define Y_MIN_POS 0
 #define Z_MIN_POS 0
 #define X_MAX_POS X_BED_SIZE + 15 // Extended max to allow the probe to reach more of the bed.
 #define Y_MAX_POS Y_BED_SIZE
@@ -1449,7 +1449,7 @@
 #if EITHER(AUTO_BED_LEVELING_LINEAR, AUTO_BED_LEVELING_BILINEAR)

   // Set the number of grid points per dimension.
+  #define GRID_MAX_POINTS_X 5
-  #define GRID_MAX_POINTS_X 3
   #define GRID_MAX_POINTS_Y GRID_MAX_POINTS_X

   // Probe along the Y axis, advancing X after each column
@@ -1683,8 +1683,8 @@
 #define PREHEAT_1_FAN_SPEED   255 // Value from 0 to 255

 #define PREHEAT_2_LABEL       "PLA"
+#define PREHEAT_2_TEMP_HOTEND 200
+#define PREHEAT_2_TEMP_BED     60
-#define PREHEAT_2_TEMP_HOTEND 190
-#define PREHEAT_2_TEMP_BED     55
 #define PREHEAT_2_FAN_SPEED   255 // Value from 0 to 255

 #define PREHEAT_3_LABEL       "ABS"
```

**Comment by smooreace** [comment]
So you didnt bake in your probe offset?

**Reply by cristian-rivera** [reply]
> So you didnt bake in your probe offset?
> […](#)
> On Sat, Jun 5, 2021 at 5:50 PM Cristian Rivera ***@***.***> wrote: I was playing around with different configurations today and decided to build my own firmware with my machine specific settings baked in. When loading up both the firmware from the official repo releases and mine, after choosing "Home All Axis's", the probe ends up in different locations! With my home offsets defined in the code, the probe ended up further back and to the right on the board whereas with the same exact home offsets set on the LCD, the probe was closer to the front / left of the bed. Is this expected behavior? - Front / left is the area on the machine above the SD card slot These were my only changes compared to https://github.com/Jyers/Marlin/blob/Ender_3_V2_Extensible_UI/Configuration%20Files/E3V2%20Templates/BLTouch-3x3/Configuration.h // Printer's unique ID, used by some programs to differentiate between machines. // Choose your own or use a service like https://www.uuidgenerator.net/version4 @@ -800,7 +800,7 @@ * Override with M92 * X, Y, Z, E0 [, E1[, E2...]] */ +#define DEFAULT_AXIS_STEPS_PER_UNIT { 80, 80, 400, 139 } -#define DEFAULT_AXIS_STEPS_PER_UNIT { 80, 80, 400, 93 } /** * Default Max Feed Rate (mm/s) @@ -1237,8 +1237,8 @@ #define Y_BED_SIZE 230 // Travel limits (mm) after homing, corresponding to endstop positions. +#define X_MIN_POS -5 +#define Y_MIN_POS -10 -#define X_MIN_POS 0 -#define Y_MIN_POS 0 #define Z_MIN_POS 0 #define X_MAX_POS X_BED_SIZE + 15 // Extended max to allow the probe to reach more of the bed. #define Y_MAX_POS Y_BED_SIZE @@ -1449,7 +1449,7 @@ #if EITHER(AUTO_BED_LEVELING_LINEAR, AUTO_BED_LEVELING_BILINEAR) // Set the number of grid points per dimension. + #define GRID_MAX_POINTS_X 5 - #define GRID_MAX_POINTS_X 3 #define GRID_MAX_POINTS_Y GRID_MAX_POINTS_X // Probe along the Y axis, advancing X after each column @@ -1683,8 +1683,8 @@ #define PREHEAT_1_FAN_SPEED 255 // Value from 0 to 255 #define PREHEAT_2_LABEL "PLA" +#define PREHEAT_2_TEMP_HOTEND 200 +#define PREHEAT_2_TEMP_BED 60 -#define PREHEAT_2_TEMP_HOTEND 190 -#define PREHEAT_2_TEMP_BED 55 #define PREHEAT_2_FAN_SPEED 255 // Value from 0 to 255 #define PREHEAT_3_LABEL "ABS" — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub <#967>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AIPE4Q6ARE2OO4JAMR6ANQLTRKS3FANCNFSM46FDTJSA> .

That didn't need to be changed, the default `#define NOZZLE_TO_PROBE_OFFSET { -44, -7, 0 }` was correct, only the home offset changed because I installed a MicroSwiss Direct Drive!

**Comment by smooreace** [comment]
Nozzle to probe offset has to be changed because its wrong for all machines
except those that run stock hot ends/coolers. and genuine Creality BL Touch

**Reply by cristian-rivera** [reply]
In my case the offsets are correct. The nozzle to probe measured with a caliper are -44/-7, the same values in the configuration, so I don't think that is the problem. 

Even if those were incorrect, they remained constant between compiling my own firmware with the changes described above and using the precompiled firmware from the releases page.

My issue is that setting a home offset hard coded and compiling vs setting them on the LCD using an official release produce different results. No other parameters between the two configs deviate. I copied the exact steps from the GitHub workflow config in this repo to compile mine.

**Comment by cristian-rivera** [comment, marked as answer]
I think I have an idea as to what is going on but I still need to validate it.

When running the official Jyers release, I had my offsets set on the LCD directly. Then I compiled my build with the offsets baked in and flashed.

Could it be that the offsets set on the printer directly and the ones set in the code are adding together?

I haven't dug deep yet in to how you implemented home offset setting on the machine, but does this sound like a possibility @Jyers ? Is setting home offsets on the machine decoupled from those two variables in the code base?

**Reply by Jyers** [reply]
Yes, I do believe these values would stack. Because while the home offsets is a persistent workspace offset applied to the position, I believe the min positions are more so "baked into" the coordinate space. Try testing with only one or the other and you should see the correct results. 

As a test you can also use M503 after a new flash to check if those configuration options set any home offsets, but like I said, I am fairly sure they do not.
